### PR TITLE
Implement w3rc CLI to look up providers for a CID

### PR DESCRIPTION
**Pull request by masih**
Implement a CLI with `find` command that looks up the list of providers
for a given CID. The command takes

Add `testscript`s for the CLI that assert expected error messages in
negative cases, and a positive case via a mock indexer server.

Relates to:
 - https://github.com/ipfs-shipyard/w3rc/issues/6

### CODE REVIEW

File: cmd/w3rc/find.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/ipfs-shipyard/w3rc/contentrouting"
+	"github.com/ipfs-shipyard/w3rc/contentrouting/delegated"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/urfave/cli/v2"
+)
+
+var findCommand = &cli.Command{
+	Name:      "find",
+	Usage:     "Finds a list of providers for a given CID",
```
[the remark this answers is not in the document]
Comment (masih) [reply cmd/w3rc/find.go:16]: The "lookup" in description of #6 made me think we want a find/lookup command in there too.

Would we not want the CLI to list providers for a CID and allow the user to choose which one to download from?

[the remark this answers is not in the document]
Comment (willscott) [reply cmd/w3rc/find.go:16]: the policy interface/module should automatically choose the provider. Ideally this isn't complexity the user has to care about.

[the remark this answers is not in the document]
Comment (masih) [reply cmd/w3rc/find.go:16]: OK, that makes sense. Closing this PR for now since without the find command it doesn't do much else.

**Comment by masih** [comment]
Closing for now until retrieval functionality is in place to implement a functioning CLI. Note, `find` is not needed as a separate top-level command.

**Pull request closed without merging**